### DISCUSSION TITLE
mpfs-hal-coremark (file error, missing file).

### DISCUSSION THREAD

**Discussion by serdat12** [polarfire-soc/polarfire-soc-documentation#260] [Bare metal embedded software] [answered]
Hello there,

I want to benchmark test. There is even a project for icicle kit. but when I opened the project, the coremark.h file was added as an include, even though it was not.

In core_portme.c file, it is passed to the function as a parameter even though secs_ret is not a variable.

fpga_design--> libero_tcl --> When I want to port the files in the LPDDR4_TOP file to Libero, the PFSOC_MSS_C0.tcl file cannot be added because it is not found.

What do I need to do and what do I need to change in order to use this project.

**Comment by mallynch** [comment, marked as answer]
Hi,  I believe the issue with missing file is down to the fact that git submodule needs to be run. From the readme.md : To use this example you will need to obtain the coremark source files from the
Embedded Microprocessor Benchmark Consortium (EEMBC). These are placed in the
coremark directory.
This will be done automatically using gitsubmodule, once you clone the repo using
git and then run the git submodule command
**git submodule update --init --recursive\n**

**Comment by gsriker77** [comment]
For Linux, You can get it from yocto recipe from below :

https://github.com/polarfire-soc/meta-polarfire-soc-yocto-bsp/blob/master/recipes-tools/coremarkpro/coremarkpro.bb



For bare metal library, get it download from below link :

CoreMark's primary goals are simplicity and providing a method for testing only a processor's core features. For more information about EEMBC's comprehensive embedded benchmark suites, please see [www.eembc.org](http://www.eembc.org/).

https://github.com/polarfire-soc/polarfire-soc-bare-metal-library/tree/master/examples/mpfs-hal/mpfs-hal-coremark/src/middleware -> ([coremark @ 41537ea](https://github.com/eembc/coremark/tree/41537ea30b0104438b4ff993e7d349af26900acf))
